game/scripts/vscripts/Dialog.ts

```diff
@@ -144,37 +164,43 @@ class DialogController {
             }
 
             if (showNextLine && hero) {
-                this.onDialogStart(hero, dialogUnit);
+                this.onDialogStart(hero);
             }
         }
     }
 
     // Unused
-    public onDialogConfirm(source: EntityIndex, data: DialogConfirmEvent) { }
+    public onDialogConfirm(source: EntityIndex, data: DialogConfirmEvent) {}
 
     // Unused
     public onDialogConfirmExpired(
         source: EntityIndex,
         data: DialogConfirmExpireEvent
-    ) { }
+    ) {}
 }
 
-const dialogController = new DialogController()
+const dialogController = new DialogController();
 
-function playCommon(line: string, unit: CDOTA_BaseNPC, duration: number, soundName?: string) {
-    const hero = getOrError(getPlayerHero(), "Can't find player hero")
+function playCommon(
+    line: string,
+    unit: CDOTA_BaseNPC,
+    duration: number,
+    soundName?: string
+) {
+    const hero = getOrError(getPlayerHero(), "Can't find player hero");
 
-    dialogController.giveUnitDialog(unit, [{
+    dialogController.addDialogToQueue({
+        speaker: unit,
         text: line,
         advance: true,
         advanceTime: duration,
         gesture: GameActivity.DOTA_ATTACK,
         sendToAll: true,
         dialogStopsMovement: false,
-        sound: soundName
-    }])
+        sound: soundName,
+    });
 
-    dialogController.onDialogStart(hero, unit)
+    dialogController.onDialogStart(hero);
```

Comment:
Should onDialogStart be called here or when the dialog is popped from the queue?